types/wlr_surface.c

```diff
@@ -174,16 +174,12 @@ static void surface_state_finalize(struct wlr_surface *surface,
 		}
 	}
 
-	if (state->buffer_resource != NULL) {
-		if (state->viewport.has_dst) {
-			state->width = state->viewport.dst_width;
-			state->height = state->viewport.dst_height;
-		} else {
-			surface_state_viewport_src_size(state,
-				&state->width, &state->height);
-		}
+	if (state->viewport.has_dst) {
```

Comment:
If the src/dst rects are set but there's no buffer, we still need to set width/height to zero.

We can probably add a `if (state->buffer_width == 0 && state->buffer_height == 0)` check?

Reply:
Added the check.